src/object/srv_obj_migrate.c

```diff
@@ -3271,8 +3272,70 @@ migrate_obj_ult(void *data)
 		}
 	}
 
+	if (!tls->mpt_reintegrating_check) {
+		struct pool_target *target;
+		struct ds_pool *pool = tls->mpt_pool->spc_pool;
+
+		ABT_rwlock_rdlock(pool->sp_lock);
```

Comment:
Is it possible during current ULT being blocked on the lock, another ULT do the same check `if (!tls->mpt_reintegrating_check)` ?